autoarray/geometry/geometry_util.py

```diff
@@ -268,3 +269,69 @@ def scaled_coordinates_2d_from(
     )
 
     return (y_pixel, x_pixel)
+
+
+def transform_grid_2d_to_reference_frame(
+    grid_2d: np.ndarray, centre: Tuple[float, float], angle: float
+) -> np.ndarray:
+    """
+    Transform a 2D grid of (y,x) coordinates to a new reference frame.
+
+    This transformation includes:
+
+    1) A translation to a new (y,x) centre value, by subtracting the centre from every coordinate on the grid.
+    2) A rotation of the grid around this new centre, which is performed clockwise from an input angle.
+
+    Parameters
+    ----------
+    grid : ndarray
+        The 2d grid of (y, x) coordinates which are transformed to a new reference frame.
+    """
+    shifted_grid_2d = np.subtract(grid_2d, centre)
+    radius = np.sqrt(np.sum(shifted_grid_2d ** 2.0, 1))
+    theta_coordinate_to_profile = np.arctan2(
+        shifted_grid_2d[:, 0], shifted_grid_2d[:, 1]
+    ) - np.radians(angle)
+    return np.vstack(

```

Comment:
Would multiplying by radius after the vstack call have the same result but be fractionally more efficient and concise?